src/strings.cpp

```diff
@@ -903,732 +901,737 @@ static void FormatString(StringBuilder &builder, const char *str_arg, StringPara
+				case SCC_REVISION: // {REV}
+					builder += _openttd_revision;
+					break;
 
-			case SCC_RAW_STRING_POINTER: { // {RAW_STRING}
-				const char *raw_string = args.GetNextParameterString();
-				/* raw_string can be(come) nullptr when the parameter is out of range and 0 is returned instead. */
-				if (raw_string == nullptr) {
-					builder += "(invalid RAW_STRING parameter)";
+				case SCC_RAW_STRING_POINTER: { // {RAW_STRING}
+					const char *raw_string = args.GetNextParameterString();
+					/* raw_string can be(come) nullptr when the parameter is out of range and 0 is returned instead. */
+					if (raw_string == nullptr) {
+						builder += "(invalid RAW_STRING parameter)";
+						break;
+					}
+					FormatString(builder, raw_string, args);
 					break;
 				}
-				FormatString(builder, raw_string, args);
-				break;
-			}
 
-			case SCC_STRING: {// {STRING}
-				StringID string_id = args.GetNextParameter<StringID>();
-				if (game_script && GetStringTab(string_id) != TEXT_TAB_GAMESCRIPT_START) break;
-				/* It's prohibited for the included string to consume any arguments. */
-				StringParameters tmp_params(args, 0);
-				GetStringWithArgs(builder, string_id, tmp_params, next_substr_case_index, game_script);
-				next_substr_case_index = 0;
-				break;
-			}
+				case SCC_STRING: {// {STRING}
+					StringID string_id = args.GetNextParameter<StringID>();
+					if (game_script && GetStringTab(string_id) != TEXT_TAB_GAMESCRIPT_START) break;
+					/* It's prohibited for the included string to consume any arguments. */
+					StringParameters tmp_params(args, game_script ? args.GetDataLeft() : 0);
+					GetStringWithArgs(builder, string_id, tmp_params, next_substr_case_index, game_script);
+					next_substr_case_index = 0;
+					break;
+				}
 
-			case SCC_STRING1:
-			case SCC_STRING2:
-			case SCC_STRING3:
-			case SCC_STRING4:
-			case SCC_STRING5:
-			case SCC_STRING6:
-			case SCC_STRING7: { // {STRING1..7}
-				/* Strings that consume arguments */
-				StringID string_id = args.GetNextParameter<StringID>();
-				if (game_script && GetStringTab(string_id) != TEXT_TAB_GAMESCRIPT_START) break;
-				uint size = b - SCC_STRING1 + 1;
-				if (game_script && size > args.GetDataLeft()) {
-					builder += "(too many parameters)";
-				} else {
-					StringParameters sub_args(args, size);
-					GetStringWithArgs(builder, string_id, sub_args, next_substr_case_index, game_script);
+				case SCC_STRING1:
+				case SCC_STRING2:
+				case SCC_STRING3:
+				case SCC_STRING4:
+				case SCC_STRING5:
+				case SCC_STRING6:
+				case SCC_STRING7: { // {STRING1..7}
+					/* Strings that consume arguments */
+					StringID string_id = args.GetNextParameter<StringID>();
+					if (game_script && GetStringTab(string_id) != TEXT_TAB_GAMESCRIPT_START) break;
+					uint size = b - SCC_STRING1 + 1;
+					if (game_script && size > args.GetDataLeft()) {
+						builder += "(too many parameters)";
+					} else {
+						StringParameters sub_args(args, size);
```

Comment:
```suggestion
						StringParameters sub_args(args, game_script ? args.GetDataLeft() : size);
```
because here they may also use the wrong STRINGN.

Reply:
Hmm, I figured if they are using `STRINGn` they are probably doing it right... but I guess it won't hurt.

Reply:
This mode of String parameters modifies the offset of the parent by the size given, so this isn't valid.
The size and offset adjustment need different values.

Reply:
oh right, don't touch `sub_args` but use `game_script ? args : sub_args` in the following call.

Reply:
Hmm doesn't seem correct either.

Reply:
If you do it that way you'll have to correct the offset afterwards.